ssl/tls13_enc.c

```diff
@@ -688,10 +725,15 @@ int tls13_change_cipher_state(SSL_CONNECTION *s, int which)
  err:
     if ((which & SSL3_CC_EARLY) != 0) {
         /* We up-refed this so now we need to down ref */
+        if ((EVP_CIPHER_flags(cipher) & EVP_CIPH_FLAG_AEAD_CIPHER) == 0)
+            ssl_evp_md_free(mac_md);
+        EVP_MD_CTX_free(mdctx);
         ssl_evp_cipher_free(cipher);
     }
     OPENSSL_cleanse(key, sizeof(key));
     OPENSSL_cleanse(secret, sizeof(secret));
+    if (ivlen > EVP_MAX_IV_LENGTH)
```

Comment:
Perhaps `iv != iv_intern` is a more robust check?

Reply:
Yes, I agree.